### PR TITLE
XSIAM - missing dependencies when installing packs

### PR DESCRIPTION
**Pull request by kobymeir**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/CIAC-6880

## Description
XSIAM - missing dependencies when installing packs

### CODE REVIEW

**Comment by kobymeir** [comment]
https://code.pan.run/xsoar/content/-/pipelines/5563478

**Comment by daryakoval** [comment]
@kobymeir The changes not solving the issue described in the ticket.

File: Tests/Marketplace/search_and_install_packs.py [hunk 1]
```diff
@@ -372,13 +372,23 @@ def install_packs(client: demisto_client,
...
+                    if len(packs_to_install) == 1:
+                        # If we only have one pack, and it failed, there is noting else to try, exiting.
+                        raise Exception(f"Got [{WLM_TASK_FAILED_ERROR_CODE}] error code - Modeling rules and Dataset validations "
+                                        f"failed. Please look at GCP logs to understand why it failed.") from ex
+
+                    # We've multiple packs to install, retrying one pack at a time.
+                    logging.debug(f"Failed to install packs:{packs_to_install}, trying one pack at a time, with only one attempt")
+                    for pack_to_install in packs_to_install:
+                        SUCCESS_FLAG &= install_packs(client, host, [pack_to_install], request_timeout, sleep_interval,
+                                                      attempts_count=1)
```
Comment (daryakoval) [inline Tests/Marketplace/search_and_install_packs.py:391]: Why retry only if WLM_TASK_FAILED_ERROR_CODE happens and not change it in the prepared batch with dependencies? 

Here it looks like plaster for only specific issues, but changing it when creating a batch with dependencies may prevent future bugs.

Comment (kobymeir) [reply Tests/Marketplace/search_and_install_packs.py:391]: it's no different from handling "malformed_ids", only here we don't know which one failed, so we try them one-by-one.

To add it to the batches will cause multiple installs of the sub-dependencies which is not efficient. 

The optimal solution is to get the exact error from the server on which pack failed.

File: Tests/Marketplace/search_and_install_packs.py [hunk 2]
```diff
@@ -372,13 +372,23 @@ def install_packs(client: demisto_client,
 
                     # We've more attempts, retrying without tho malformed packs.
                     SUCCESS_FLAG = False
-                    logging.error(f"Unable to install malformed packs: {malformed_ids}, retrying without them.")
                     packs_to_install = [pack for pack in packs_to_install if pack['id'] not in malformed_ids]
+                    if packs_to_install:
+                        logging.error(f"Unable to install malformed packs: {malformed_ids}, retrying without them.")
+                    else:
+                        raise Exception(f"Unable to install malformed packs: {malformed_ids}, no other packages left to install")
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:379]: packages?!?
```suggestion
                        raise Exception(f"Unable to install malformed packs: {malformed_ids}, no other packs left to install")
```

File: Tests/Marketplace/search_and_install_packs.py [hunk 3]
```diff
@@ -372,13 +372,23 @@ def install_packs(client: demisto_client,
...
 
                 if (error_ids := get_error_ids(ex.body)) and WLM_TASK_FAILED_ERROR_CODE in error_ids:
-                    # If we got this error code, it means that the modeling rules are not valid, exiting install flow.
-                    raise Exception(f"Got [{WLM_TASK_FAILED_ERROR_CODE}] error code - Modeling rules and Dataset validations "
-                                    f"failed. Please look at GCP logs to understand why it failed.") from ex
+                    if len(packs_to_install) == 1:
+                        # If we only have one pack, and it failed, there is noting else to try, exiting.
```
Comment (tkatzir) [inline Tests/Marketplace/search_and_install_packs.py:383]:
```suggestion
                        # If we only have one pack, and it failed, there is nothing else to try, exiting.
```

**Review by tkatzir — APPROVED** [review]
Nice!
Approved with minor comments.

**Comment by kobymeir** [comment]
test...
```
--- a/Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py
+++ b/Packs/AzureSentinel/Integrations/AzureSentinel/AzureSentinel.py
@@ -1895,6 +1895,7 @@ def create_and_update_alert_rule_command(client: AzureSentinelClient, args: Dict
     )
 
 
+
 def main():
```

**Pull request closed without merging**